### ISSUE TITLE
Crash on add address

### ISSUE THREAD

**Issue by LordDarkHelmet** [duality-solutions/Dynamic#377]
### Describe the issue
Crash in the windows QT wallet.

### Steps to reproduce
1. Close 2.3.5 wallet 
2. Upgrade to 2.4.0 wallet (Windows AVX2 using the installer)
3. Let 2.4.0 fully sync
4. Click File->Receiving addresses
5. Click New
6. Type in a new label
7. Press OK.
8. Full crash. See log below.

Log file Extract, Run 1:
```
2019-08-31 22:58:40 CDynodeMan::ProcessVerifyBroadcast -- verified Dynode xxxxxxxxxxxxxxxxxx-0 for addr xxx.xxx.xxx.xxx:33300
2019-08-31 22:59:35 keypool reserve 146
2019-08-31 22:59:35 GUI: Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.
```

Log File Extract, Run 2:
```
2019-08-31 23:05:55 ERROR: CDynodePaymentVote::CheckSignature -- Got bad Dynode payment signature, dynode=xxxxxxxxxxxxxxxxxx-0, error: Keys don't match: pubkey=xxxxxxxxxxxxxxxx, pubkeyFromSig=xxxxxxxxxxxxxxxxxxx, hash=xxxxxxxxxxxxxxxxxxxxx, vchSig=xxxxxxxx+xxxxxxxx/xxxxxxxxxx+xxxxxxxx=
2019-08-31 23:05:55 ERROR: CDynodePaymentVote::CheckSignature -- Got bad Dynode payment signature, dynode=yyyyyyyyyyyyyyyyyy-0, error: Keys don't match: pubkey=yyyyyyyyyyyyyyy, pubkeyFromSig=yyyyyyyyyyyyyyy, hash=yyyyyyyyyyyyyyyyyyyy, vchSig=yyyyyyyy+yyyyyyyyyyyyyyyyy=
2019-08-31 23:07:19 keypool reserve 146
2019-08-31 23:07:19 GUI: Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.
```
### Expected behavior (tell us what should happen)
a new address is created

### Actual behavior (tell us what happens instead)
application crash

### What platform are you using (Linux, Windows, Mac)
Windows 10

**Comment by SvenKercher** [comment]
U need to unlock the encrypted wallet once, this allows the wallet to read the HDseed and create the ed25519 keys for ur addresses and for the keypool , also getnewaddress command/clickable at qt use a address from keypool, so without unlock the wallet u will always that error until u unlocked it one time

**Comment by LordDarkHelmet** [comment]
I am aware of what I "should" do. The issue is that it is possible for me to do this. If someone does not read or know about your instructions, they will run into the same issue. This should be fixed. There is a history of changing the wallet.dat file properties that lead to incompatibility. It is probable that another change will happen in the future. 

This issue should be trapped, then the user should be prompted for their password to resolve the issue.

**Comment by LordDarkHelmet** [comment]
Or you could just require the user to perform corrective action before they are allowed to interact with the wallet.

**Comment by AmirAbrams** [comment]
@LordDarkHelmet, thank you for reporting the issue with the reproduction steps. 

We are working on adding warnings to the UI/daemon that prevent the segfault.  The fix will be included in v2.4.1.

**Comment by LordDarkHelmet** [comment]
Sounds great! Thank you

**Comment by Duality-CDOO** [comment]
Fixed #379

**Comment by LordDarkHelmet** [comment]
Looks good. Screenshots were nice.
